File: Leetcode/leetcode-0026-RemoveDuplicatesFromSortedArray-[array]-[easy]/Solution.two-pointers.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let i = 0,
    j = 0;
  let curVal = "unknown";
  while (i < nums.length) {
    if (nums[i] !== curVal) {
      curVal = nums[i];
      nums[j] = nums[i];
      j++;
    }

    i++;
  }

  return j;
};

File: Leetcode/leetcode-0026-RemoveDuplicatesFromSortedArray-[array]-[easy]/Solution.two-pointers.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        j = 0
        curVal = 'unknown'
        while (i < len(nums)):
            if (nums[i] != curVal):
                curVal = nums[i]
                nums[j] = nums[i]
                j = j + 1
            
            i = i + 1

        return j